pupil_src/shared_modules/video_capture/ndsi_backend.py

```diff
@@ -408,76 +434,82 @@ def re_build_ndsi_menu(self):
         ui_elements.append(
             ui.Selector(
                 "selected_host",
                 self,
                 selection=host_sel,
                 labels=host_sel_labels,
-                setter=view_host,
+                setter=self.view_host,
                 label="Remote host",
             )
         )
 
         self.menu.extend(ui_elements)
+        self.add_auto_select_button()
+
         if not self.selected_host:
             return
         ui_elements = []
 
         host_menu = ui.Growing_Menu("Remote Host Information")
         ui_elements.append(host_menu)
 
-        def source_selection_list():
-            default = (None, "Select to activate")
-            # self.poll_events()
-            sources = [default] + [
-                (s["sensor_uuid"], s["sensor_name"])
-                for s in self.network.sensors.values()
-                if (
-                    s["sensor_type"] == "video" and s["host_uuid"] == self.selected_host
-                )
-            ]
-            return zip(*sources)
-
-        def activate(source_uid):
-            if not source_uid:
-                return
-            settings = {
-                "frame_size": self.g_pool.capture.frame_size,
-                "frame_rate": self.g_pool.capture.frame_rate,
-                "source_id": source_uid,
-            }
-            self.activate_source(settings)
-
-        src_sel, src_sel_labels = source_selection_list()
+        src_sel, src_sel_labels = self.source_selection_list()
         host_menu.append(
             ui.Selector(
                 "selected_source",
                 selection=src_sel,
                 labels=src_sel_labels,
                 getter=lambda: None,
-                setter=activate,
+                setter=self.activate,
                 label="Source",
             )
         )
 
         self.menu.extend(ui_elements)
 
+    def activate(self, source_uid):
+        if not source_uid:
+            return
+        settings = {
+            "frame_size": self.g_pool.capture.frame_size,
+            "frame_rate": self.g_pool.capture.frame_rate,
+            "source_id": source_uid,
+        }
+        self.activate_source(settings)
+
+    def auto_select_manager(self):
+        super().auto_select_manager()
+        self.notify_all(
+            {
+                "subject": "backend.ndsi_host_selected",
+                "selected_host": self.selected_host,
+                "delay": 0.4,
+            }
+        )
+
+    def auto_activate_source(self):
+        cam_id = self.cam_selection_lut[self.g_pool.process]
+
+        src_sel, src_sel_labels = self.source_selection_list()
+        logger.info(src_sel)
+        logger.info(src_sel_labels)
+        if src_sel is None or src_sel_labels is None:
+            logger.warning("here?")
+            return
+
+        source_id = [
+            src_sel[i] for i, lab in enumerate(src_sel_labels) if cam_id in lab
+        ]
+
+        if len(source_id) != 1:
```

Comment:
Could it happen that there are more than one s.t. this should rather be `if len(source_id) == 0`?

Reply:
Not sure. Would any single host would have more than one headset connected?

We can have 
```python
        if len(source_id) == 0:
            return

        source_id = source_id[0]
```
i.e. choosing the first camera that matches the criteria.

Reply:
Just ignore this. See comments below. I'm resolving this.